src/super_gradients/training/models/segmentation_models/unet/unet_decoder.py

```diff
@@ -0,0 +1,182 @@
+class UpFactorBlock(AbstractUpFuseBlock):
+    """
+    Ignore Skip features, simply apply upsampling and ConvBNRelu layers.
+    """
+    def __init__(self,
+                 in_channels: int,
+                 skip_channels: int,
+                 out_channels: int,
+                 up_factor: int,
+                 mode: str,
+                 num_repeats: int,
+                 **kwargs
+                 ):
+        super().__init__(in_channels=in_channels, skip_channels=0, out_channels=out_channels)
+        self.up_path = make_upsample_module(scale_factor=up_factor, upsample_mode=mode, align_corners=False)
+
+        self.last_convs = nn.Sequential(
+            ConvBNReLU(in_channels, out_channels, kernel_size=3, padding=1, bias=False),
+            nn.Sequential(*[ConvBNReLU(out_channels, out_channels, kernel_size=3, padding=1, bias=False) for _ in range(num_repeats - 1)]),
+        )
+
+    def forward(self, x, skip):
+        x = self.up_path(x)
+        return self.last_convs(x)
+
+
+class UpCatBlock(AbstractUpFuseBlock):
+    """
+    Fuse features with concatenation and followed Convolutions.
+    """
+    def __init__(self,
+                 in_channels: int,
+                 skip_channels: int,
+                 out_channels: int,
+                 up_factor: int,
+                 mode: str,
+                 num_repeats: int,
+                 **kwargs
+                 ):
+        super().__init__(in_channels=in_channels, skip_channels=skip_channels, out_channels=out_channels)
+        self.up_path = make_upsample_module(scale_factor=up_factor, upsample_mode=mode, align_corners=False)
+        self.last_convs = nn.Sequential(
+            ConvBNReLU(in_channels + skip_channels, out_channels, kernel_size=3, padding=1, bias=False),
+            nn.Sequential(*[ConvBNReLU(out_channels, out_channels, kernel_size=3, padding=1, bias=False) for _ in range(num_repeats - 1)]),
+        )
+
+    def forward(self, x, skip):
+        x = self.up_path(x)
+        x = torch.cat([x, skip], dim=1)
+        return self.last_convs(x)
+
+
+class UpBlockType(Enum):
+    UP_FACTOR = (0, UpFactorBlock)
+    UP_CAT = (1, UpCatBlock)
+
+    def __init__(self,
+                 block_id: int,
+                 up_block_cls: Type[AbstractUpFuseBlock]):
+        self.block_id = block_id
+        self.up_block_cls = up_block_cls
+
+    @staticmethod
+    def from_block_id(block_id: int):
+        if block_id == UpBlockType.UP_FACTOR.block_id:
+            return UpBlockType.UP_FACTOR
+        elif block_id == UpBlockType.UP_CAT.block_id:
+            return UpBlockType.UP_CAT
+        raise NotImplementedError(f"block_id: {block_id} is not a valid option.")
+
+
+class Decoder(nn.Module):
+    def __init__(self,
+                 skip_channels_list: List[int],
+                 up_block_repeat_list: List[int],
+                 skip_expansion: float,
+                 decoder_scale: float,
+                 up_block_types: List[Union[UpBlockType, int]],
```

Comment:
why do you allow int here? (if because of NAS then remove this)

Reply:
Removed [f33631f](https://github.com/Deci-AI/super-gradients/pull/401/commits/f33631f226d77345abd9db0c38b369015efc6e17)